code/modules/projectiles/guns/projectile.dm

```diff
@@ -43,7 +44,7 @@
 
 //loads the argument magazine into the gun
 /obj/item/weapon/gun/projectile/proc/LoadMag(var/obj/item/ammo_storage/magazine/AM, var/mob/user)
-	if(istype(AM, text2path(mag_type)) && !stored_magazine)
+	if(istype(AM, text2path(mag_type)) && !stored_magazine && !istype(AM, mag_type_restricted))
```

Comment:
change this to work with a list, eg:
loop over contents and try an istype for each one